app/build.gradle.kts

```diff
@@ -131,15 +132,19 @@ android {
 
     compileOptions {
         isCoreLibraryDesugaringEnabled = true
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        sourceCompatibility = JavaVersion.VERSION_17
```

Comment:
Yes, I agree. Ideally we should be on LTS stuff, but sourceCompatibility is one of those things which just works and has not hindered any development so far.

Changing it is just a little scary because there is more to it than compiling. We have to make sure this does not fuck any extension stuff and that android 5 works as expected. Dex stuff is especially weird on Android 5 and can easily go unnoticed in changes such as these. For all I know *some* android 5 devices do not support Java 17 code. 

To my understanding we currently compile using JDK 17, but that compiler makes sure the code is backwards compatible with the old ass java version. We are not losing on any cool features, just compiler optimizations and such.

So a change like this warrants proper motivation and testing, even if it seems trivial.

It's kinda like this xkcd
![image](https://github.com/recloudstream/cloudstream/assets/46196380/3124353a-52e9-4b2e-8c15-52564b5226a0)

Reply:
man, i love ur explanations 😙, i have tested extension stuff in android 12,12.1 and 13 , will have to test on low version especially 5 and 6, but will have to find testers